sky/skylet/job_lib.py

```diff
@@ -190,6 +200,68 @@ def set_status(job_id: int, status: JobStatus) -> None:
         _set_status_no_lock(job_id, status)
 
 
+class JobScheduler:
+    """Class for general job scheduler"""
+
+    def queue(self, job_id: int, cmd: str) -> None:
+        _CURSOR.execute('INSERT INTO pending_jobs VALUES (?,?,?,?)',
+                        (job_id, cmd, 0, int(time.time())))
+        _CONN.commit()
+        set_status(job_id, JobStatus.PENDING)
+        self.run_next_if_possible()
+
+    def _get_job_status(self, job_id: int):
+        status = list(
+            _CURSOR.execute(f'SELECT status FROM jobs WHERE job_id={job_id}'))
+        if len(status) == 0:
+            return False
+        return JobStatus(status[0][0])
+
+    def _run_job(self, job_id: int, run_cmd: str):
+        _CURSOR.execute(
+            f'UPDATE pending_jobs SET submit=1 WHERE job_id={job_id!r}')
+        _CONN.commit()
+        subprocess.Popen(run_cmd, shell=True, stdout=subprocess.DEVNULL)
+
+    def remove_job(self, job_id: str):
+        with filelock.FileLock(_get_lock_path(job_id)):
+            self.remove_job_no_lock(job_id)
+
+    def remove_job_no_lock(self, job_id: str) -> None:
+        _CURSOR.execute(f'DELETE FROM pending_jobs WHERE job_id={job_id!r}')
+        _CONN.commit()
+
+    def run_next_if_possible(self) -> None:
+        raise NotImplementedError
```

Comment:
How about we move the `update_status` here?

```suggestion
    def run_next_if_possible(self) -> None:
        job_owner = getpass.getuser()
        update_status(job_owner)
        self._run_next_if_possilbe(self)
    def _run_next_if_possilbe(self):
        raise NotImplementedError
```